### PR TITLE
DropConnect

### PR DESCRIPTION
**Pull request by wonglkd**
To test

### CODE REVIEW

File: ddrop/layers.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    def __init__(self, layer, prob=1., **kwargs):
+        self.prob = prob
+        self.layer = layer
+        if 0. < self.prob < 1.:
+            self.uses_learning_phase = True
+        super(DropConnect, self).__init__(**kwargs)
+
+    def call(self, x, mask=None):
+        if 0. < self.prob < 1.:
+            self.W = K.in_train_phase(K.dropout(self.W, self.prob), self.W)
```
Comment (wonglkd) [inline ddrop/layers.py]: This is an approximation, because it scales it by (1-p) - see [keras/backend/theano_backend.py#L1303](https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L1303).

To properly implement DropConnect, I was thinking that one cheap hack would be to multiply it by (1-p).

Also, you probably want to make use of `K.in_test_phase` to do the averaging thing during prediction... the list of operators available can be found at https://keras.io/backend/. You can also use TensorFlow operators directly if you don't mind breaking compatibility with Theano.

File: ddrop/layers.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from keras.layers import Dense
+from keras.wrappers import Wrapper
+import keras.backend as K
+
+
+class DropConnectDense(Dense):
```
Comment (wonglkd) [inline ddrop/layers.py]: can be removed once we verify that `DropConnect` works

**Comment by derekkhu** [comment]
Dropout

https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L1274
https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L2464

**Comment by derekkhu** [comment]
looks good to me
